File: 0547-number-of-provinces/0547-number-of-provinces.java
class Solution {
    //constructor
    public boolean[] visited;
    public Solution(){
        
        //ArrayList<Boolean> visited = new ArrayList<>();
        //Queue<Integer> queue = new Queue<>();
        
    }
    public int findCircleNum(int[][] isConnected) {
        //1. Determine the length (number of cities)
        
        int m = isConnected.length;
        this.visited = new boolean[m];
        int counter = 0;
        //2. For 
        for(int i = 0;i<m;i++){
            if(this.visited[i] == false){
                counter++;
                this.dfs(i, isConnected);
                
            }
        }
        
        return counter;
    }
    
    public void dfs(int i, int[][] isConnected){
        
        //1. Check the necessary conditions 
        if (this.visited[i] == true){
            return;
        }
        
        //2. Process the current cell 
        this.visited[i] = true;
        
        //3. Recursive dfs calls
        for(int k = 0;k<isConnected[i].length;k++){
            //check if the connection exists 
            if(isConnected[i][k] == 1){
                this.dfs(k, isConnected);
            }
        }
        
    }
}

File: 0547-number-of-provinces/0547-number-of-provinces.py
class Solution(object):
    def __init__(self):
        self.visited = []
    def findCircleNum(self, isConnected):
        """
        :type isConnected: List[List[int]]
        :rtype: int
        """
        m = len(isConnected)
        n = len(isConnected[0])
        self.visited = [0] * m
        counter = 0
        # for loop for everything and do a DFS 
        for i in range(m):
                if self.visited[i] == 0:
                    counter+=1
                    self.dfs(i, isConnected)
        
        return counter
    
    def dfs(self, i, isConnected):
        # Check all the necessary condition 
        if self.visited[i] == 1:
            return 
        
        # Process the current cell 
        self.visited[i] = 1
        
        # Call the DFS function as needed
        for k in range(len(isConnected[i])):
            if(isConnected[i][k] == 1):
                self.dfs(k, isConnected)
                    
        